### PR TITLE
Added passing of additional metadata about element to tool, and tests.

### PR DESCRIPTION
**Pull request by gspencergoog**
In implementing my custom tool, I realized that I needed to know more information about the input in order to generate unique IDs for my output files, and in order to have the names make sense, it's useful to have them use metadata from dartdoc. I added things that seemed like they would be generally useful to external tools, but there could be more (notably, I didn't include the source code that the comment is attached to, but I could).

### CODE REVIEW

File: lib/src/tool_runner.dart [hunk 1]
```diff
@@ -86,18 +88,28 @@ class ToolRunner {
...
+      argsWithInput.add(newArg);
     }
 
     argsWithInput = toolArgs + argsWithInput;
     final commandPath = argsWithInput.removeAt(0);
     String commandString() => ([commandPath] + argsWithInput).join(' ');
     try {
-      var result = Process.runSync(commandPath, argsWithInput);
+      var result = Process.runSync(commandPath, argsWithInput,
+          environment: {}..addAll(Platform.environment)..addAll(envWithInput));
```
Comment (jcollins-g) [inline lib/src/tool_runner.dart]: nit: i think the default of includeParentEnvironment: true covers including the current environment.

Comment (gspencergoog) [reply lib/src/tool_runner.dart]: Ooh, good point.  Fixed.

**Comment by jcollins-g** [comment]
Looks like you need a rebuild of test package docs?

**Comment by gspencergoog** [comment]
Yeah, I did it once, but then I needed to do it again...and rebase onto upstream/master, etc. This one should pass.

**Comment by gspencergoog** [comment]
Actually, it's not that simple. I need to have my dummy tool in the tests sanitize it's output so that it doesn't contain the absolute path to the temp file...

**Comment by gspencergoog** [comment]
I had moved it to the test code, to keep the regexs in one place, but now I remember why I hadn't done that before.

**Pull request merged**